### PR TITLE
Changed tab to space conversion

### PR DESCRIPTION
**Pull request by QuentinC**
Tabs are not always 4 spaces large. It completes the line to the 4th
character.

### CODE REVIEW

**Comment by QuentinC** [comment]
You can see the result here:
[Gitblit](https://git.o2r.fr/blob/~quentin!weatherStation.git/master/WeatherStation!src!main!java!fr!o2r!iniParser!IniFile.java) (lines 14-16)

File: src/main/java/com/gitblit/utils/StringUtils.java [hunk 1]
```diff
@@ -106,12 +108,17 @@ public static String escapeForHtml(String inStr, boolean changeSpace, int tabLen
...
+				for (int j = 0; j < tabLength - l; j++) {
 					retStr.append("&nbsp;");
 				}
+				l = -1;
 			} else {
 				retStr.append(inStr.charAt(i));
 			}
+			
+			l = (l + 1) % tabLength;
+			if (inStr.charAt(i) == '\n')
```
Comment (gitblit) [inline src/main/java/com/gitblit/utils/StringUtils.java]: Would you wrap this `if` with curly braces?

**Comment by gitblit** [comment]
This looks like a nice fix.  Could you write a unit test or two to verify that it works like expected so in the future we don't accidentally break your improvement?

**Comment by QuentinC** [comment]
No problem. I'll do this on sunday !

Do you see a way to implement this ?


Would testing a tab for let's say, the 50 first positions would be enough ?

Quentin Canel
Ingénieur ISAE-ENSMA
Option Informatique et Avionique
40 Allée Henri Guillaumet
77350 Le Mée sur Seine
France
06.70.37.34.65


________________________________
De : James Moger <notifications@github.com>
Envoyé : mercredi 20 avril 2016 13:44
À : gitblit/gitblit
Cc : Quentin Canel; State change
Objet : Re: [gitblit/gitblit] Changed tab to space conversion (#1065)


This looks like a nice fix. Could you write a unit test or two to verify that it works like expected so in the future we don't accidentally break your improvement?

-

**Comment by QuentinC** [comment]
I've added a simple tab->space conversion testing.

**Comment by QuentinC** [comment]
Hello,

Dou you plan to merge this for the 1.8.0 release ?

Thanks !

**Comment by flaix** [comment]
Closed as merged.

**Pull request closed without merging**